test/integration/controller/jobs/deployment/deployment_webhook_test.go

```diff
@@ -0,0 +1,131 @@
+
+	ginkgo.When("with pod integration enabled", ginkgo.Ordered, ginkgo.ContinueOnFailure, func() {
+		ginkgo.BeforeAll(func() {
+			fwk = &framework.Framework{
+				CRDPath:     crdPath,
+				WebhookPath: webhookPath,
+			}
+			cfg = fwk.Init()
+
+			discoveryClient, err := discovery.NewDiscoveryClientForConfig(cfg)
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+			serverVersionFetcher = kubeversion.NewServerVersionFetcher(discoveryClient)
+			err = serverVersionFetcher.FetchServerVersion()
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+
+			ctx, k8sClient = fwk.RunManager(cfg, managerSetup(
+				nil,
+				jobframework.WithManageJobsWithoutQueueName(false),
+				jobframework.WithKubeServerVersion(serverVersionFetcher),
+				jobframework.WithIntegrationOptions(corev1.SchemeGroupVersion.WithKind("Pod").String(), &configapi.PodIntegrationOptions{
+					PodSelector: &metav1.LabelSelector{},
+					NamespaceSelector: &metav1.LabelSelector{
+						MatchExpressions: []metav1.LabelSelectorRequirement{
+							{
+								Key:      "kubernetes.io/metadata.name",
+								Operator: metav1.LabelSelectorOpNotIn,
+								Values:   []string{"kube-system", "kueue-system"},
+							},
+						},
+					},
+				}),
+			))
+		})
+		ginkgo.BeforeEach(func() {
+			ns = &corev1.Namespace{
+				ObjectMeta: metav1.ObjectMeta{
+					GenerateName: "deployment-",
+				},
+			}
+			gomega.Expect(k8sClient.Create(ctx, ns)).To(gomega.Succeed())
+		})
+
+		ginkgo.AfterEach(func() {
+			gomega.Expect(util.DeleteNamespace(ctx, k8sClient, ns)).To(gomega.Succeed())
+		})
+		ginkgo.AfterAll(func() {
+			fwk.Teardown()
+		})
+
+		ginkgo.When("The queue-name label is set", func() {
+			var (
+				deployment *appsv1.Deployment
+				lookupKey  types.NamespacedName
+			)
+
+			ginkgo.BeforeEach(func() {
+				deployment = testingdeployment.MakeDeployment("deployment-with-queue-name", ns.Name).
+					Queue("user-queue").
+					Obj()
+				lookupKey = types.NamespacedName{Name: deployment.Name, Namespace: deployment.Namespace}
+			})
+
+			ginkgo.It("Should inject queue name to pod template labels", func() {
+				gomega.Expect(k8sClient.Create(ctx, deployment)).Should(gomega.Succeed())
+
+				createdDeployment := &appsv1.Deployment{}
+				gomega.Eventually(func() error {
+					return k8sClient.Get(ctx, lookupKey, createdDeployment)
+				}, util.Timeout, util.Interval).Should(gomega.Succeed())
+
+				fmt.Printf("createdDeployment: %v\n", createdDeployment.Spec.Template.Labels)
+				gomega.Expect(createdDeployment.Spec.Template.Labels[constants.QueueLabel]).To(
+					gomega.Equal("user-queue"),
+					"Queue name should be injected to pod template labels",
+				)
+			})
+
+			ginkgo.It("Should not allow to modify queue label", func() {
+				gomega.Expect(
+					k8sClient.Patch(
+						ctx,
+						deployment,
+						client.MergeFrom(testingdeployment.MakeDeployment("deployment-with-queue-name", ns.Name).
+							Queue("updated-queue").
+							Obj())),
+				).To(gomega.HaveOccurred())
```

Comment:
Instead of using the patch operation, could you get the Deployment and update the queue-label?